test/test-functions

```diff
@@ -1117,16 +1117,25 @@ check_asan_reports() {
 }
 
 save_journal() {
+    # Default to always saving journal
+    local save="yes"
+
+    if [ "${TEST_SAVE_JOURNAL}" = "no" ]; then
+        save="no"
+    elif [ "${TEST_SAVE_JOURNAL:0:4}" = "fail" ] && [ "$2" = "0" ]; then
```

Comment:
I am not familiar with shell scripting, but why `:0:4` is added here?

Reply:
my intention was so the variable would still work if set to 'failed' or 'failing' instead of 'fail', the trailing :0:4 uses the substring from chars 0-3 (start at char 0, 4 chars length).

I could add a comment or I could just remove the :0:4 and have a direct compare if you think that's better.

Reply:
I think we should not support various values starting from "fail". Supporting only "fail" (or "failed" is better?) is ok.

Reply:
ack, pushed update to just check for 'no' and 'fail'

i also merged the 2 commits into one